ui/page/exchange/create_order_page.go

```diff
@@ -610,6 +617,10 @@ func (pg *CreateOrderPage) Layout(gtx C) D {
 		overlaySet = true
 	}
 
+	if !overlaySet && !pg.inited {
+		pg.initPage()
+	}
```

Comment:
This will always be false because you are sett `pg.inited` to true in the `OnNavigatedTo()` and i dont see any where it is set back to false.

Reply:
We need to execute this once. If the page was already in the stack, we never run OnNavigatedTo() which does the first init.

Reply:
When ever a page is navigated to, the `OnNavigatedTo()` is always called the last time i checked. if the plan is to only initialise this once, i will recommend moving it to the `NewCreateOrderPage()`

Reply:
> When ever a page is navigated to, the `OnNavigatedTo()` is always called the last time i checked. if the plan is to only initialise this once, i will recommend moving it to the `NewCreateOrderPage()`

Not really. For example navigating to the create order view will execute OnNavigatedTo the first time but won’t when we visit the general setting view from there and use the back arrow icon on the settings page.